File: tests/machine/x/go/if_then_else.go
//go:build ignore

package main

import (
	"fmt"
)

func main() {
	x := 12
	_ = x
	msg := func() string {
		if x > 10 {
			return "yes"
		} else {
			return "no"
		}
	}()
	fmt.Println(msg)
}

File: tests/machine/x/lua/if_then_else.lua
function __print(...)
    local args = {...}
    local parts = {}
    for i,a in ipairs(args) do
        if a ~= nil and a ~= '' then parts[#parts+1] = tostring(a) end
    end
    print(table.concat(parts, ' '))
end
x = 12
msg = (function()
  if (x > 10) then
    return "yes"
  else
    return "no"
  end
end)()
__print(msg)
